types/env.go

```diff
@@ -8,8 +8,9 @@ package types
 //
 // Env are json encoded to a byte slice before passing to the wasm contract.
 type Env struct {
-	Block    BlockInfo    `json:"block"`
-	Contract ContractInfo `json:"contract"`
+	Block       BlockInfo        `json:"block"`
+	Contract    ContractInfo     `json:"contract"`
+	Transaction *TransactionInfo `json:"transaction_info,omit_empty"`
```

Comment:
q: Would this be empty for simulations/ queries? In the "spike" I did not make a different and defaulted to `0`.

Reply:
Yes. Also empty if the contract was called in a cron job or such. Thus optional.

If this was directly triggered by a TX in a block (DeliverTx), then this should be filled in.
This allows contacts to store some unique reference (requested by Secret Network, but may be useful in other use cases to identify or order tx)